File: src/main/java/com/janrain/backplane/server2/AuthSession.java
/*
 * Copyright 2012 Janrain, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.janrain.backplane.server2;

import com.janrain.backplane.common.DateTimeUtils;
import com.janrain.backplane.common.ExternalizableCore;
import com.janrain.commons.message.MessageException;
import com.janrain.commons.message.MessageField;

import java.util.*;

/**
 * @author Johnny Bufu
 */
public class AuthSession extends ExternalizableCore {

    // - PUBLIC

    /**
     * Empty default constructor for AWS to use
     */
    public AuthSession() { }
    
    public AuthSession(String authUser, String cookie) throws MessageException {
        Map<String,String> data = new LinkedHashMap<String, String>();
        data.put(Field.AUTH_USER.getFieldName(), authUser);
        data.put(Field.COOKIE.getFieldName(), cookie);
        data.put(Field.EXPIRES.getFieldName(), DateTimeUtils.ISO8601.get().format(new Date(System.currentTimeMillis() + AUTH_SESSION_TIMEOUT_SECONDS * 1000)));
        super.init(cookie, data);
    }

    @Override
    public String getIdValue() {
        return get(Field.COOKIE);
    }

    @Override
    public Set<? extends MessageField> getFields() {
        return EnumSet.allOf(Field.class);
    }

    public static enum Field implements MessageField {

        // - PUBLIC

        COOKIE,
        AUTH_USER,
        EXPIRES;

        @Override
        public String getFieldName() {
            return name();
        }

        @Override
        public boolean isRequired() {
            return true;
        }

        @Override
        public void validate(String value) throws MessageException {
            if (isRequired()) validateNotBlank(name(), value);
        }
    }

    // - PRIVATE

    private static final long serialVersionUID = -8416685362228294237L;

    private static final long AUTH_SESSION_TIMEOUT_SECONDS = 3600l;
}

File: src/main/scala/com/janrain/backplane/server2/model/AuthSession.scala
package com.janrain.backplane.server2.model

import com.janrain.backplane.common.DateTimeUtils
import java.util.Date
import com.janrain.backplane.common.model.{MessageField, MessageFieldEnum, Message}

/**
 * @author Johnny Bufu
 */
class AuthSession(data: Map[String,String]) extends Message(data, AuthSessionFields.values) {

  def this(authUser: String, cookie: String) = this(
    Map( AuthSessionFields.COOKIE.name -> cookie,
         AuthSessionFields.AUTH_USER.name -> authUser,
         AuthSessionFields.EXPIRES.name
           -> DateTimeUtils.ISO8601.get().format(new Date(System.currentTimeMillis() + AuthSession.AUTH_SESSION_TIMEOUT_SECONDS * 1000))
    ))

  def idField = AuthSessionFields.COOKIE
}

object AuthSession {
  private[model] final val AUTH_SESSION_TIMEOUT_SECONDS: Long = 3600L
}

object AuthSessionFields extends MessageFieldEnum {

  type AuthSessionField = EnumVal

  sealed trait EnumVal extends Value with MessageField {
    def required = true
  }

  val COOKIE = new AuthSessionField { def name = "cookie" }

  val AUTH_USER = new AuthSessionField { def name = "auth_user" }

  val EXPIRES = new AuthSessionField { def name = "expires" }
}
